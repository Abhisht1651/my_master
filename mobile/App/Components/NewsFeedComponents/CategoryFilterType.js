// @flow
import React, { useState } from 'react'
import { Image, TouchableWithoutFeedback, View, Text } from 'react-native'
import PropTypes from 'prop-types'
import styles from '../Styles/ScrollableFeedComponentStyles/CategoryFilterTypeStyles'
import { Images } from '../../Themes'

type Props = {
  title: string,
  image: Image,
  selected: Array<string>,
  activityChange: PropTypes.func
}

const CategoryFilterType = (props: Props) => {
  const { title, image, selected, activityChange } = props
  const [value, setValue] = useState(!!selected.find((e) => e === title))
  const clickHandler = () => {
    setValue(!value)
    activityChange(value, title)
  }
  return (
    <TouchableWithoutFeedback onPress={clickHandler}>
      <View style={styles.cardItems}>
        <View style={styles.iconContainer}>
          <Image source={image} style={styles.iconStyles} />
        </View>
        <View style={styles.itemText}>
          <Text style={styles.text}>{title}</Text>
        </View>
        <View style={styles.checkboxContainer}>
          <View style={styles.checkBox}>
            {selected.find((e) => e === title) && (
              <View style={styles.checkedBox}>
                <Image source={Images.checkmark} style={styles.checkmark} />
              </View>
            )}
          </View>
        </View>
      </View>
    </TouchableWithoutFeedback>
  )
}

export default CategoryFilterType
