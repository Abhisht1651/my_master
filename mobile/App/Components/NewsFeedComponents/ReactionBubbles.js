// @flow
import React, { useState } from 'react'
import { View, FlatList, Image } from 'react-native'
import { Images } from '../../Themes'
import styles from '../Styles/ScrollableFeedComponentStyles/ReactionBubblesStyles'
import Storyfromfriends1 from '../StoriesFromFriends/StoriesFromFriends1'
import Storyfromfriends2 from '../StoriesFromFriends/StoriesFromFriends2'

type Props = {
  item: {
    id: string,
    seen: boolean,
    image1: Image,
    image2: Image,
    name: string,
    image: Image
  }
}

// user reaction
const user = [
  {
    id: '1',
    image: Images.profilePicture1
  }
]

// friend reaction
const bubbles = [
  {
    id: 'bd7acbea-c1b1-46c2-aed5-3ad53abb28ba',
    seen: false,
    image1: Images.flagPicture,
    image2: Images.profilePicture3,
    name: 'Amelia'
  },
  {
    id: '3ac68afc-c605-48d3-a4f8-fbd9a97f63',
    seen: true,
    image1: Images.flagPicture,
    image2: Images.profilePicture3,
    name: 'Amelia'
  },
  {
    id: '58694a0f-3da1-471f-bd96-1455e29d72',
    seen: true,
    image1: Images.flagPicture,
    image2: Images.profilePicture3,
    name: 'Amelia'
  },
  {
    id: '58694a0f-3da1-471f-bd9645571e29d72',
    seen: false,
    image1: Images.flagPicture,
    image2: Images.profilePicture3,
    name: 'Amelia'
  },
  {
    id: '58694a0f-3da1-471f-96-145571e29d72',
    seen: false,
    image1: Images.flagPicture,
    image2: Images.profilePicture3,
    name: 'Amelia'
  },
  {
    id: '58694a0f-a1-471f-bd96-145571e29d72',
    seen: false,
    image1: Images.flagPicture,
    image2: Images.profilePicture3,
    name: 'Amelia'
  },
  {
    id: '58694a0a1-471f-bd96-145571e29d72',
    seen: false,
    image1: Images.flagPicture,
    image2: Images.profilePicture3,
    name: 'Amelia'
  },
  {
    id: '58694a0f-3da1-471f-bd96571e29d72',
    seen: false,
    image1: Images.flagPicture,
    image2: Images.profilePicture3,
    name: 'Amelia'
  }
]

// Functionality for changing seen to true on opening reaction is not done yet. Data has an element named *seen*.

export const ReactionBubbles = () => {
  const [stories] = useState(user.concat(bubbles))
  const renderstories = ({ item }: Props) => {
    if (item.id === '1') {
      return (
        <View style={styles.margin1}>
          <Storyfromfriends1 image={item.image} />
        </View>
      )
    }
    return (
      <View style={styles.margin}>
        <Storyfromfriends2
          image1={item.image1}
          image2={item.image2}
          seen={item.seen}
          name={item.name}
        />
      </View>
    )
  }
  return (
    <View style={styles.bubbles}>
      <FlatList
        data={stories}
        renderItem={renderstories}
        keyExtractor={(image) => image.id}
        horizontal
        showsHorizontalScrollIndicator={false}
        style={styles.marginTop}
      />
    </View>
  )
}
