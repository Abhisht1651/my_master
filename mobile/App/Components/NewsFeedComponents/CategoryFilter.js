// @flow

import React from 'react'
import { View, Text, TouchableOpacity, SafeAreaView, FlatList } from 'react-native'
import PropTypes from 'prop-types'
import Images from '../../Themes/Images'
import styles from '../Styles/ScrollableFeedComponentStyles/CategoryFilterStyles'
import CategoryFilterType from './CategoryFilterType' // helper component for category component

type Props = {
  activityChange: PropTypes.func,
  applyChange: PropTypes.func,
  selected: Array<string>
}
const CategoryFilterCard = (props: Props) => {
  const { activityChange, applyChange, selected } = props
  const data = [
    {
      id: 1,
      title: 'World',
      image: Images.world
    },
    {
      id: 2,
      title: 'National',
      image: Images.national
    },
    {
      id: 3,
      title: 'Sports',
      image: Images.sports
    },
    {
      id: 4,
      title: 'Science',
      image: Images.science
    },
    {
      id: 5,
      title: 'Entertainment',
      image: Images.entertainment
    },
    {
      id: 6,
      title: 'Technology',
      image: Images.technology
    },
    {
      id: 7,
      title: 'Others',
      image: Images.others
    }
  ]

  return (
    <SafeAreaView style={styles.popUpCard}>
      <View style={styles.categoryFilter}>
        <Text style={styles.categoryFilterText}>Category Filter</Text>
      </View>
      <FlatList
        contentContainerStyle={styles.flatList}
        data={data}
        renderItem={({ item }) => (
          <CategoryFilterType
            title={item.title}
            image={item.image}
            id={item.id}
            selected={selected}
            activityChange={activityChange}
          />
        )}
        keyExtractor={(item) => item.id.toString()}
      />
      <View style={styles.options}>
        <View>
          {selected.length > 0 && (
            <TouchableOpacity activeOpacity={0.6}>
              <Text style={styles.selectedText}>{selected.length} Selected</Text>
            </TouchableOpacity>
          )}
        </View>

        <TouchableOpacity activeOpacity={0.6} onPress={applyChange}>
          <Text style={styles.clearSelectionText}>Clear Selection</Text>
        </TouchableOpacity>
      </View>

      <View style={styles.footer}>
        <View style={styles.bottomLine} />
      </View>
    </SafeAreaView>
  )
}

export default CategoryFilterCard
