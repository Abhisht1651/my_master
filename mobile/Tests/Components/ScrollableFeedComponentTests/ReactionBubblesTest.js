import React from 'react'
import { shallow } from 'enzyme'
import { FlatList } from 'react-native'
import { ReactionBubbles } from '../../../App/Components/NewsFeedComponents/ReactionBubbles'
// Test for ScrollableFeed (snapshot and props)
describe('ReactionBubbles', () => {
  let wrapper
  beforeEach(() => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    wrapper = shallow(<ReactionBubbles />) // no compile-time error
  })
  it('ReactionBubbles component renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should flatlist return keyExtractor correctly', () => {
    expect(wrapper.find(FlatList)).toHaveLength(1)
  })
})
