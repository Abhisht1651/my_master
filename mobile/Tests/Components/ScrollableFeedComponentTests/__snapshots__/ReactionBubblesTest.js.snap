// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionBubbles ReactionBubbles component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderBottomColor": "#E6ECED",
      "borderBottomWidth": 1,
      "height": 251.86197916666669,
      "width": 750,
    }
  }
>
  <FlatList
    data={
      Array [
        Object {
          "id": "1",
          "image": Object {
            "default": "",
          },
        },
        Object {
          "id": "bd7acbea-c1b1-46c2-aed5-3ad53abb28ba",
          "image1": Object {
            "default": "",
          },
          "image2": Object {
            "default": "",
          },
          "name": "Amelia",
          "seen": false,
        },
        Object {
          "id": "3ac68afc-c605-48d3-a4f8-fbd9a97f63",
          "image1": Object {
            "default": "",
          },
          "image2": Object {
            "default": "",
          },
          "name": "Amelia",
          "seen": true,
        },
        Object {
          "id": "58694a0f-3da1-471f-bd96-1455e29d72",
          "image1": Object {
            "default": "",
          },
          "image2": Object {
            "default": "",
          },
          "name": "Amelia",
          "seen": true,
        },
        Object {
          "id": "58694a0f-3da1-471f-bd9645571e29d72",
          "image1": Object {
            "default": "",
          },
          "image2": Object {
            "default": "",
          },
          "name": "Amelia",
          "seen": false,
        },
        Object {
          "id": "58694a0f-3da1-471f-96-145571e29d72",
          "image1": Object {
            "default": "",
          },
          "image2": Object {
            "default": "",
          },
          "name": "Amelia",
          "seen": false,
        },
        Object {
          "id": "58694a0f-a1-471f-bd96-145571e29d72",
          "image1": Object {
            "default": "",
          },
          "image2": Object {
            "default": "",
          },
          "name": "Amelia",
          "seen": false,
        },
        Object {
          "id": "58694a0a1-471f-bd96-145571e29d72",
          "image1": Object {
            "default": "",
          },
          "image2": Object {
            "default": "",
          },
          "name": "Amelia",
          "seen": false,
        },
        Object {
          "id": "58694a0f-3da1-471f-bd96571e29d72",
          "image1": Object {
            "default": "",
          },
          "image2": Object {
            "default": "",
          },
          "name": "Amelia",
          "seen": false,
        },
      ]
    }
    disableVirtualization={false}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "height": "100%",
        "top": 29.528645833333336,
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
